src/main/java/com/rsmart/kuali/kfs/cr/document/service/impl/GlTransactionServiceImpl.java

```diff
@@ -168,11 +168,11 @@ private void generateGlPendingTransaction(PaymentGroup paymentGroup, String fina
             
             Boolean relieveLiabilities = paymentGroup.getBatch().getCustomerProfile().getRelieveLiabilities();
             if ((relieveLiabilities != null) && (relieveLiabilities.booleanValue()) && paymentAccountDetail.getPaymentDetail().getFinancialDocumentTypeCode() != null) {
-                OffsetDefinition offsetDefinition = SpringContext.getBean(OffsetDefinitionService.class).getByPrimaryId(glPendingTransaction.getUniversityFiscalYear(), glPendingTransaction.getChartOfAccountsCode(), paymentAccountDetail.getPaymentDetail().getFinancialDocumentTypeCode(), glPendingTransaction.getFinancialBalanceTypeCode());
+                OffsetDefinition offsetDefinition = SpringContext.getBean(OffsetDefinitionService.class).getActiveByPrimaryId(glPendingTransaction.getUniversityFiscalYear(), glPendingTransaction.getChartOfAccountsCode(), paymentAccountDetail.getPaymentDetail().getFinancialDocumentTypeCode(), glPendingTransaction.getFinancialBalanceTypeCode()).orElse(null);
                 
                 glPendingTransaction.setAccountNumber(accountNbr);
                 glPendingTransaction.setChartOfAccountsCode(finCoaCd);
-                glPendingTransaction.setFinancialObjectCode(offsetDefinition != null ? offsetDefinition.getFinancialObjectCode() : finObjectCode);
+                glPendingTransaction.setFinancialObjectCode(ObjectUtils.isNotNull(finObjectCode) ? offsetDefinition.getFinancialObjectCode() : finObjectCode);

```

Comment:
I think this null check should examine `offsetDefinition`, not `finObjectCode`.

Reply:
fixing